test/CMakeLists.txt

```diff
@@ -42,6 +49,27 @@ macro(onedpl_add_test test_source_file)
     set_target_properties(${_test_name} PROPERTIES CXX_EXTENSIONS NO)
 
     add_test(NAME ${_test_name} COMMAND ${CMAKE_CURRENT_BINARY_DIR}/${_test_name})
+
+    set(build_types "release;debug")
```

Comment:
Even if you prefer to keep separate handling for different build types I don't think it makes sense to have `build_types` and iterate through them.

Consider the following approach:
```cmake

string(TOLOWER "${CMAKE_BUILD_TYPE}" _build_type_in_lower)
if (DEFINED ${_test_name}_timeout_${_build_type_in_lower})
    set(${_test_name}_timeout ${_test_name}_timeout_${_build_type_in_lower})
elseif (DEFINED ${_test_name}_timeout_release)
    # Apply default (release) timeout
    set(${_test_name}_timeout ${_test_name}_timeout_release)
endif()

if (DEFINED ${_test_name}_timeout)
    message(STATUS "Timeout for ${_test_name} is set to ${${_test_name}_timeout}, it overrides user-defined timeout for CTest")
    set_tests_properties(${_test_name} PROPERTIES TIMEOUT ${${_test_name}_timeout})
endif()

```

Reply:
If you'll choose unified approach, then just keep the last `if`

Reply:
Such scenario isn't workable, because instead of timeout value test name is setup. In this case, copy of the variable containing the same content with a different name wont created, as it was expected.
For example:
```cmake
set(subtract_with_carry_std_template_test.pass_timeout_release "900")
set(${_test_name}_timeout ${_test_name}_timeout_release)
message(${_test_name}_timeout_release)
message(${_test_name}_timeout)
message(${${_test_name}_timeout_release})
message(${${_test_name}_timeout})
```
Output:
```bash
subtract_with_carry_std_template_test.pass_timeout_release
subtract_with_carry_std_template_test.pass_timeout
900
subtract_with_carry_std_template_test.pass_timeout_release
```

However, I have simplified the change, pls, review one more time.

Reply:
Yes, I forgot to wrap right part of `set`s with `${}`, it should work if fix it. But I'm ok with your implementation